src/audio/chain_dma.c

```diff
@@ -284,17 +286,29 @@ static int chain_task_start(struct comp_dev *dev)
 		}
 	}
 
-	pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+	ret = schedule_task_init_ll(&cd->chain_task, SOF_UUID(chain_dma_uuid),
+				    SOF_SCHEDULE_LL_TIMER, SOF_TASK_PRI_HIGH,
+				    chain_task_run, cd, 0, 0);
+	if (ret < 0) {
+		comp_err(dev, "chain_task_start(), ll task initialization failed");
+		goto error_task;
+	}
 
 	ret = schedule_task(&cd->chain_task, 0, 0);
-	if (ret) {
-		chain_host_stop(dev);
-		chain_link_stop(dev);
-		goto error;
+	if (ret < 0) {
+		comp_err(dev, "chain_task_start(), ll schedule task failed");
+		goto error_task;
```

Comment:
how about this case - will the task be free on this path or not? Looks like it has to be freed here